src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -191,6 +191,22 @@ public void setLevel(int level) {
      * <p>
      * The current time will be used if the entry has no set modification time.
      *
+     * @apiNote When writing a directory entry, the STORED compression method
+     * should be used and the size and CRC-32 values should be set to 0:
+     *
+     * {@snippet lang="java" :
+     *     ZipEntry e = ...;
```

Comment:
I initially used a directory as an example: `ZipEntry e = new ZipEntry("dir/")`, but then @AlanBateman  suggested to make it more generic:

https://mail.openjdk.org/pipermail/core-libs-dev/2023-March/101686.html

> For the note then you might want to change it to "ZipEntry e = ..." because the reader see the trailing slash after dir so it is obviously a directory. 

I agree valid code would be preferrable. Not sure how to make it valid though, since the `isDirectory` part suggests the  path is unknown. Could an undefined local variable work?

```java
ZipEntry e = new ZipEntry(name);
```

Or perhaps with a comment:

```java
ZipEntry e = new ZipEntry(name); // name could be a file or directory
```

Reply:
I opted for `ZipEntry e = new ZipEntry(entryName)` for now.